7957: Remove unnecessary argument brave_branding_path_component

mariospr opened the pull request:
This allows us to drop build-config-chrome_build.gni.patch and the
part in build/features.gni setting its value for non iOS builds,
since this will now be set in via brave-browser's config.js file.

Resolves brave/brave-browser#7957


## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

N/A

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

mkarolin commented on patches/build-config-chrome_build.gni.patch:
```diff
@@ -1,20 +0,0 @@
- # Copyright 2015 The Chromium Authors. All rights reserved.
- # Use of this source code is governed by a BSD-style license that can be
- # found in the LICENSE file.
-+import("//brave/build/config/brave_build.gni")
```
If removing this line from that patch causes all the new patches below, then I think we'd want to keep this line and just remove the branding line below.

mariospr replied:
I had exactly the same thoughts while doing this, but then I figured that since the lines added everywhere else go into patches that already existed, then perhaps it did make some sense in the end, since we would be getting rid of one patch file after all. No strong opinion, though, looking forward to seeing what others think.

mariospr replied:
@mkarolin @bridiver Hi again! Sorry for pushing, but since this PR has sit here for a few days already and it's blocking https://github.com/brave/brave-browser/pull/10758 as well, I believe a ping is in order.

@bridiver Could you clarify whether you'd be ok with this change even if it implies modifying several other -already existing- patches? (yet it allows us to drop one, see my comment above)

Thanks in advance!

bridiver replied:
@mariospr we're not trying to minimize the number of files with patches, we are trying to minimize the total number of patches and therefore seems to be moving in the wrong direction. I agree with @mkarolin that it doesn't make sense to remove this here only to have to add it to a bunch of other places. We specifically added it here so it would be available pretty much everywhere.

mariospr commented:
> please undo the import("//brave/build/config/brave_build.gni") for the reasons explained here https://github.com/brave/brave-core/pull/6091/files#r458969780

@bridiver Thanks for the review, I've addressed your review comments but in a different PR, since created this one in my own fork. See the new one in https://github.com/brave/brave-core/pull/6172
